### DISCUSSION TITLE
Client channel pool/manager or thick clients?

### DISCUSSION THREAD

**Discussion by kyle-mccarthy** [hyperium/tonic#1469] [Q&A]
Are there any implementations of a grpc channel manager? Ideally, once the underlying TCP connection becomes saturated, under the hood, the manager would spawn new connections based on predefined conditions.

My use case focuses on bidi streams, where the server can handle a higher load (at the same endpoint) but the connection has reached its limits.

**Comment by LucioFranco** [comment]
There is the balance option but right now we don't have any automatic behavior. The balance feature should add more connections corresponding with load.
